### PR TITLE
Need to use correct email address in notifications

### LINKED ISSUE #494: Need to use correct email address in notifications
@MisterJames  While working on issue #473 it came to my attention that the previously developed code that generates email notifications when new users sign up for an activity (NotifyAdminForTaskSignupStatusChange.cs) and a task status change takes place (NotifyAdminForTaskSignupStatusChange.cs) sources the admin email address from the **Campaign.Organizer** property.  This property is not being populated anywhere in the app and I was instructed for #473 to get the email address from the **Campaign.CampaignContacts** collection.  If this is indeed correct, I have made the changes to the above two referenced routines to use the email address from **Campaign.CampaignContacts** and will submit a PR with those changes.

Is **Campaign.Organizer** a dead property?  If so, should it be removed to avoid confusion?

### PR DESCRIPTION
**Pull request by mmoore99**
Fix #494

### CODE REVIEW

**Comment by mmoore99** [comment]
any idea as to why the build keeps failing?

**Comment by dpaquette** [comment]
Looks like an npm error that I have not seen before. I will try to investigate today

**Comment by tonysurma** [comment]
@MisterJames can someone look over this PR before I merge?

File: AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForActivitySignup.cs [hunk 1]
```diff
@@ -27,13 +27,16 @@ public void Handle(VolunteerInformationAdded notification)
...
             var campaign = _context.Campaigns
-                .Include(c => c.Organizer)
+                .Include(c => c.CampaignContacts).ThenInclude(cc => cc.Contact)
                 .Single(c => c.Id == activity.CampaignId);
             var link = $"View activity: http://{_options.Value.SiteBaseUrl}/Admin/Activity/Details/{activity.Id}";
 
             var subject = $"A volunteer has signed up for {activity.Name}";
 
-            if (campaign.Organizer != null)
+            var campaignContact = campaign.CampaignContacts.SingleOrDefault(tc => tc.ContactType == (int)ContactTypes.Primary);
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForActivitySignup.cs]: Though it's not functionally possible right now with user workflows, handle the possibility that `CampaignContacts` could be null. `var campaignContact = campaign.CampaignContacts?.SingleOrDefault(...)`

File: AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForTaskSignupStatusChange.cs [hunk 2]
```diff
@@ -36,12 +36,9 @@ public void Handle(TaskSignupStatusChanged notification)
             var activity = task.Activity;
             var campaign = activity.Campaign;
 
-            //Prefer activity organizer email; fallback to campaign organizer email
-            var adminEmail = taskSignup.Task.Activity.Organizer?.Email;
-            if (string.IsNullOrWhiteSpace(adminEmail))
-            {
-                adminEmail = taskSignup.Task.Activity.Campaign.Organizer?.Email;
-            }
+            var campaignContact = campaign.CampaignContacts.SingleOrDefault(tc => tc.ContactType == (int)ContactTypes.Primary);
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForTaskSignupStatusChange.cs]: Handle the possibility that `CampaignContacts` could be null. `var campaignContact = campaign.CampaignContacts?.SingleOrDefault(...)`

**Comment by mmoore99** [comment]
Submitted changes as requested by @mheggeseth in PR review.

**Comment by mheggeseth** [comment]
Merge it. :shipit:

**Pull request merged**